### PR TITLE
[DW-2133] Added button component

### PR DESCRIPTION
**Pull request by Deven-Rawat**

### CODE REVIEW

**Comment by ShaneAhern-IRE** [comment]
Could you please rebase the commits into a single commit

**Comment by Deven-Rawat** [comment]
> Could you please rebase the commits into a single commit

Rebase done.

File: site/components/src/main/java/uk/nhs/digital/common/components/ButtonComponent.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            request.setAttribute("button3Link", button3Link);
+        }
+
+        final String heading = componentInfo.getHeader();
+        final String alignment = componentInfo.getAlignment();
+
+        request.setAttribute("heading", heading);
+        request.setAttribute("alignment", alignment);
+
+        System.out.println(button1Link);
```
Comment (ShaneAhern-IRE) [inline site/components/src/main/java/uk/nhs/digital/common/components/ButtonComponent.java:47]: Those System.out.println should be removed

File: repository-data/webfiles/src/main/resources/site/freemarker/common/button.ftl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<#ftl output_format="HTML">
+
+<#include "../include/imports.ftl">
+<#include "macro/metaTags.ftl">
+
+<#-- Add meta tags -->
+<@metaTags></@metaTags>
```
Comment (ShaneAhern-IRE) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/button.ftl:7]: I don't think the metaTags are needed on the catalog component.
Worth checking if they're already added to the DOM by an ftl higher up.

File: site/components/src/main/java/uk/nhs/digital/common/components/ButtonComponent.java (hunk 1 above)
Comment (Deven-Rawat) [reply site/components/src/main/java/uk/nhs/digital/common/components/ButtonComponent.java]: This has been removed

File: repository-data/webfiles/src/main/resources/site/freemarker/common/button.ftl (hunk 2 above)
Comment (Deven-Rawat) [reply repository-data/webfiles/src/main/resources/site/freemarker/common/button.ftl]: This has been removed

**Review by ShaneAhern-IRE — APPROVED** [review]
Changes look good to me

**Pull request merged**